llama-index-core/llama_index/core/evaluation/retrieval/metrics.py

```diff
@@ -12,45 +12,51 @@
 
 
 class HitRate(BaseRetrievalMetric):
-    """Hit rate metric."""
+    """Hit rate metric: Compute the proportion of matches between retrieved documents and expected documents."""
 
     metric_name: str = "hit_rate"
 
     def compute(
         self,
-        query: Optional[str] = None,
         expected_ids: Optional[List[str]] = None,
         retrieved_ids: Optional[List[str]] = None,
-        expected_texts: Optional[List[str]] = None,
-        retrieved_texts: Optional[List[str]] = None,
-        **kwargs: Any,
     ) -> RetrievalMetricResult:
         """Compute metric."""
-        if retrieved_ids is None or expected_ids is None:
-            raise ValueError("Retrieved ids and expected ids must be provided")
-        is_hit = any(id in expected_ids for id in retrieved_ids)
-        return RetrievalMetricResult(
-            score=1.0 if is_hit else 0.0,
-        )
+        if (
+            retrieved_ids is None
+            or expected_ids is None
+            or not retrieved_ids
+            or not expected_ids
+        ):
+            raise ValueError("Both retrieved ids and expected ids must be provided")
 
+        expected_set = set(expected_ids)
+        hits = sum(1 for doc_id in retrieved_ids if doc_id in expected_set)
+        score = hits / len(expected_ids) if expected_ids else 0.0
 
-class MRR(BaseRetrievalMetric):
-    """MRR metric."""
+        return RetrievalMetricResult(score=score)
 
-    metric_name: str = "mrr"
+
+class RR(BaseRetrievalMetric):
+    """Reciprocal Rank (RR): Calculates the reciprocal rank of the first, and only the first, relevant retrieved document.
+    returns 0 if no relevant retrieved docs are found.
+    """
+
+    metric_name: str = "rr"
 
     def compute(
         self,
-        query: Optional[str] = None,
         expected_ids: Optional[List[str]] = None,
         retrieved_ids: Optional[List[str]] = None,
-        expected_texts: Optional[List[str]] = None,
```

Comment:
I'm not entirely sure if these can be removed. Did you check if removing these args still satisfies the `compute` method signature for `BaseRetrievalMetric`?

Reply:
Yep you are correct. I did some research and it would violate the Liskov substitution principle.

When making the changes, I will ensure the method signatures are kept the same

Thanks for highlighting this

Reply:
Thanks!